File: Converter/app/src/test/java/apps/stp/converter/main/FloatingNumberTest.java
package apps.stp.converter.main;

import apps.stp.converter.util.Misc;
import org.junit.Test;

public class Kekejfkef {
    @Test
    public void main(){
//        String resultNumber = convertNumber("-17.875", "10", "16");
//        String resultNumber = convertNumber("AA.4", "16", "10");
        String resultNumber = convertNumber("010100111.10101", "2", "10");
        System.out.println(resultNumber);
    }

   public String convertNumber(String inputNumber, String inputNumberSystem, String resultNumberSystem) {
        if(!inputNumber.contains(Misc.DOT_SYMBOL)) {
            String convertedNumber = convertNumberNoDot(inputNumber, inputNumberSystem, resultNumberSystem);
            return convertedNumber.toUpperCase();
        }
        String [] splitNumber = inputNumber.split("\\" + Misc.DOT_SYMBOL);
        String leftPart = convertNumberNoDot(splitNumber[0], inputNumberSystem, resultNumberSystem);
        String rightPart = getNumberAfterDot(splitNumber[1], inputNumberSystem, resultNumberSystem);
        return leftPart + "." + rightPart.toUpperCase();
    }

    private String convertNumberNoDot(String inputNumber, String inputNumberSystem, String resultNumberSystem){
        String result = Integer.toString(Integer.parseInt(inputNumber, Integer.parseInt(inputNumberSystem)), Integer.parseInt(resultNumberSystem));
        return result;
    }

    private String getNumberAfterDot(String numberAfterDot, String inputNumberSystem, String resultNumberSystem){
        if(Integer.valueOf(resultNumberSystem) == 10) {
            return convertNumberNoDot(numberAfterDot, inputNumberSystem, resultNumberSystem);
        }
        int resultNumberAfterDot = (int) (Double.parseDouble(resultNumberSystem) * Double.parseDouble("0." + numberAfterDot));
        return convertNumberNoDot(String.valueOf(resultNumberAfterDot), inputNumberSystem, resultNumberSystem);
    }
}

File: Converter/app/src/test/java/apps/stp/converter/main/FloatingNumberTest.kt
package apps.stp.converter.main

import org.junit.Assert
import org.junit.Test

class FloatingNumberTest {
    @Test
    fun main() {
        val model = ConvertNumberModel();
        var expectedValue : String
        var actualValue : String

        expectedValue = "-11.E"
        actualValue = model.convertNumber("-17.875", "10", "16")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "-17.875"
        actualValue = model.convertNumber("-11.E", "16", "10")
        Assert.assertEquals(expectedValue, actualValue)


        expectedValue = "170.25"
        actualValue = model.convertNumber("AA.4", "16", "10")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "AA.4"
        actualValue = model.convertNumber("170.25", "10", "16")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "85.6875"
        actualValue = model.convertNumber("1010101.1011", "2", "10")
        Assert.assertEquals(expectedValue, actualValue)


        expectedValue = "1010101.1011"
        actualValue = model.convertNumber("85.6875", "10", "2")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "55.B"
        actualValue = model.convertNumber("1010101.1011", "2", "16")
        Assert.assertEquals(expectedValue, actualValue)


        expectedValue = "1010101.1011"
        actualValue = model.convertNumber("55.B", "16", "2")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "A.CCCCCCC"
        actualValue = model.convertNumber("10.8", "10", "16")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "10.796875"
        actualValue = model.convertNumber("A.CCCCCCC", "16", "10", "6")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "0.1101"
        actualValue = model.convertNumber("0.847", "10", "2", "4")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "0.6615"
        actualValue = model.convertNumber("0.847", "10", "8", "4")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "9A7.44"
        actualValue = model.convertNumber("4647.214", "8", "16", "4")
        Assert.assertEquals(expectedValue, actualValue)

        expectedValue = "4647.214"
        actualValue = model.convertNumber("9A7.46", "16", "8", "4")
        Assert.assertEquals(expectedValue, actualValue)
    }
}
